Python/instrumentation.c

```diff
@@ -2284,3 +2284,361 @@ PyObject *_Py_CreateMonitoringObject(void)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, target_offset };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_JUMP);
+}
+
+int
+_PyMonitoring_FireBranchEvent(PyMonitoringState *state, PyObject *codelike, int offset,
+                              PyObject *target_offset)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, target_offset };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_BRANCH);
+}
+
+int
+_PyMonitoring_FirePyThrowEvent(PyMonitoringState *state, PyObject *codelike, int offset,
+                               PyObject *exception)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, exception };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_PY_THROW);
+}
+
+int
+_PyMonitoring_FireRaiseEvent(PyMonitoringState *state, PyObject *codelike, int offset,
+                             PyObject *exception)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, exception };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_RAISE);
+}
+
+int
+_PyMonitoring_FireReraiseEvent(PyMonitoringState *state, PyObject *codelike, int offset,
+                               PyObject *exception)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, exception };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_RERAISE);
+}
+
+int
+_PyMonitoring_FireExceptionHandledEvent(PyMonitoringState *state, PyObject *codelike, int offset,
+                                        PyObject *exception)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, exception };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_EXCEPTION_HANDLED);
+}
+
+int
+_PyMonitoring_FirePyUnwindEvent(PyMonitoringState *state, PyObject *codelike, int offset,
+                                PyObject *exception)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, exception };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_PY_UNWIND);
+}
+
+int
+_PyMonitoring_FireStopIterationEvent(PyMonitoringState *state, PyObject *codelike, int offset,
+                                     PyObject *exception)
+{
+    assert(state->active);
+    PyObject *args[4] = { NULL, NULL, NULL, exception };
+    return capi_call_instrumentation(state, codelike, offset, args, 3,
+                                     PY_MONITORING_EVENT_STOP_ITERATION);
+}
+
+
+
+#ifdef Py_LIMITED_API
```

Comment:
These functions need to be always defined as they are part of the ABI.
We never compile with `Py_LIMITED_API` defined, so this code will always be ignored.

Reply:
If I always define them then it clashes with the static inline version in ./Include/monitoring.h